### DISCUSSION TITLE
Retrieving the current rate from the event

### DISCUSSION THREAD

**Discussion by ialexyi** [BottlecapDave/HomeAssistant-OctopusEnergy#840] [Q&A]
Hi,

Please advise how to get the current rate from the event - event.octopus_energy_electricity_current_day_rates to be offline. Currently there is a current rate but it relied on the Internet, isn't it. I would like to create a new entity current_offline_rate. Is it possible?

**Comment by BottlecapDave** [comment]
Hello. This integration relies on the internet. If it can't retrieve the rates, then the rate sensors will use any cached data available. This data is cached until it can retrieve updated information, HA is restarted or the integration is reloaded. The events will only fire when it can successfully retrieve the rates from OE. You can use the attributes of the event sensors to get the cached data or the `current rate` sensors to get the current rate.

**Reply by ialexyi** [reply]
Thanks Dave
